syntax = "proto3";

package ocrservice;

service OCRService {
  rpc ProcessImage(OCRRequest) returns (OCRResponse) {}
  rpc ProcessImageStream(stream OCRRequest) returns (stream OCRResponse) {}
}

message OCRRequest {
  string request_id = 1;  // Changed to string
  bytes image_data = 2;
}

message OCRResponse {
  string request_id = 1;  // Changed to string
  string text = 2;
  bool success = 3;
  string error_message = 4;
}